internal/hostgacommunicator/hostgacommunicator.go

```diff
@@ -34,18 +42,24 @@ type IVMSettingsRequestManager interface {
 }
 
 // GetVMSettings returns the VMSettings for the current machine
-func (c *HostGACommunicator) GetImmediateVMSettings(ctx *log.Context) (*VMSettings, error) {
+func (c *HostGACommunicator) GetImmediateVMSettings(ctx *log.Context, eTag string) (*ResponseData, error) {
 	ctx.Log("message", "getting request manager")
 	requestManager, err := c.vmRequestManager.GetVMSettingsRequestManager(ctx)
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not create the request manager")
 	}
 
 	ctx.Log("message", "attempting to make request with retries to retrieve VMSettings")
-	resp, err := requesthelper.WithRetries(ctx, requestManager, requesthelper.ActualSleep)
+	resp, err := requesthelper.WithRetries(ctx, requestManager, requesthelper.ActualSleep, eTag)
 	if err != nil {
 		return nil, errors.Wrapf(err, "metadata request failed with retries.")
 	}
+
+	if resp.StatusCode == http.StatusNotModified {
+		ctx.Log("message", "ETag has not changed. No need to parse response body")
```

Comment:
Yes, it will run on every call to HGAP when the ETag has not changed. Actually, let me check again the code because even if we are making long polling calls, if the ETag hasn;t changed, that will be reported back immediately and then we are going to be calling the API every second 😅